### PR TITLE
PR w issue

### PR DESCRIPTION
**Pull request by planteater**
yes this does

**This PR Addresses:**  
[COD-865 Codemark location is lost](https://teamcodestream.atlassian.net/browse/COD-865)  

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

**Comment by planteater** [comment]
I can haz replies.

**Comment by planteater** [comment]
> yes this does

hmmm

**Comment by planteater** [comment]
single 49-51

```
	dispose() {
		this._disposable && this._disposable.dispose();
	}
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L49-L51) (Lines 49-51)

File: providers/codeActionProvider.ts [hunk 1]
```diff
@@ -21,6 +21,22 @@ export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposa
...
+		const status = e.getStatus();
+		switch (status) {
+			case SessionStatus.SignedOut:
+				this._disposableSignedIn && this._disposableSignedIn.dispose();
+				break;
+
+			case SessionStatus.SignedIn:
+				this._disposableSignedIn = languages.registerCodeActionsProvider(
+					CodeStreamCodeActionProvider.selector,
+					this
```
Comment (planteater) [inline providers/codeActionProvider.ts:34]: single in cs 32-34

File: providers/codeActionProvider.ts [hunk 2]
```diff
@@ -21,6 +21,22 @@ export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposa
...
+				);
+				break;
+		}
+	}
+
 	private readonly _disposable: Disposable;
 	private _disposableSignedIn: Disposable | undefined;
```
Comment (planteater) [inline providers/codeActionProvider.ts:41]: single close to CS 40-41

**Comment by planteater** [comment]
far from cs 53-55

```
	private async onSessionStatusChanged(e: SessionStatusChangedEvent) {
		const status = e.getStatus();
		switch (status) {
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L53-L55) (Lines 53-55)

**Comment by bcanzanella** [comment]
z

```
tatic selector: DocumentSelector = [
		{ scheme: "file" },
		{ scheme: "untitled" },
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L18-L20) (Lines 18-20)

**Comment by bcanzanella** [comment]
a

```
this._disposable = Disposable.from(
			Container.session.onDidChangeSessionStatus(this.onSessionStatusChanged, this)
		);
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L44-L46) (Lines 44-46)

**Comment by bcanzanella** [comment]
43

```
	constructor() {

```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L43) (Line 43)

File: providers/codeActionProvider.ts [hunk 3]
```diff
@@ -21,6 +21,22 @@ export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposa
...
```
Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 42

Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 41 44

Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 4144

Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 4144

Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 4144

**Comment by bcanzanella** [comment]
43

```
	constructor() {

```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L43) (Line 43)

File: providers/codeActionProvider.ts (hunk 3 above)
Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 42

**Comment by bcanzanella** [comment]
4045

```
private readonly _disposable: Disposable;
	private _disposableSignedIn: Disposable | undefined;

	constructor() {
		this._disposable = Disposable.from(
			Con
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L40-L45) (Lines 40-45)

**Comment by bcanzanella** [comment]
18

```
	static selector: DocumentSelector = [

```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L18) (Line 18)

**Comment by bcanzanella** [comment]
20

```
		{ scheme: "untitled" },

```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L20) (Line 20)

File: providers/codeActionProvider.ts [hunk 4]
```diff
@@ -21,6 +21,22 @@ export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposa
 		{ scheme: "vsls" }
```
Comment (bcanzanella) [inline providers/codeActionProvider.ts:21]: 21

File: providers/codeActionProvider.ts [hunk 5]
```diff
@@ -21,6 +21,22 @@ export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposa
...
 	];
 
+	private async onSessionStarted(e: SessionStatusChangedEvent) {
+		const status = e.getStatus();
+		switch (status) {
+			case SessionStatus.SignedOut:
+				this._disposableSignedIn && this._disposableSignedIn.dispose();
+				break;
```
Comment (bcanzanella) [inline providers/codeActionProvider.ts:29]: 2629

**Comment by bcanzanella** [comment]
1724

```
export class CodeStreamCodeActionProvider implements CodeActionProvider, Disposable {
	static selector: DocumentSelector = [
		{ scheme: "file" },
		{ scheme: "untitled" },
		{ scheme: "vsls" }
	];

	private 
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L17-L24) (Lines 17-24)

File: agent/agentConnection.ts [hunk 6]
```diff
@@ -443,6 +443,13 @@ export class CodeStreamAgentConnection implements Disposable {
 			});
 		}
 
+		fetchMyBeer(streamId: string, parentPostId: string) {
```
Comment (bcanzanella) [inline agent/agentConnection.ts:446]: 446

**Comment by bcanzanella** [comment]
a

```
		text: text,
				mentionedUserIds: mentionedUserIds,
				parentPostId: parentPostId,
				codemark: codemark
			});
		}

		fetchMyBeer(streamId: string, parentPostId: string) {
			return this._connection.sendRequest(FetchPostRepliesRequestType, {
			
```
								
[agent/agentConnection.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/agent/agentConnection.ts#L439-L448) (Lines 439-448)

File: providers/codeActionProvider.ts (hunk 3 above)
Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: aaaaa

Comment (bcanzanella) [inline providers/codeActionProvider.ts:42]: 39-42

**Comment by bcanzanella** [comment]
5558

```
	switch (status) {
			case SessionStatus.SignedOut:
				this._disposableSignedIn && this._disposableSignedIn.dispose();
				break;
```
								
[providers/codeActionProvider.ts](https://github.com/planteater/hello-ny/blob/d167923490665df0654f44287e27c96413e8805d/providers/codeActionProvider.ts#L55-L58) (Lines 55-58)

**Pull request closed without merging**